awx/main/access.py

```diff
@@ -540,6 +545,73 @@ def can_unattach(self, obj, sub_obj, relationship, *args, **kwargs):
         return super(UserAccess, self).can_unattach(obj, sub_obj, relationship, *args, **kwargs)
 
 
+class OauthApplicationAccess(BaseAccess):
+    '''
+    I can read, change or delete OAuth applications when:
```

Comment:
`of the user of the application` is a confusing bit of the sentence that's used a few times in this section. Can we clarify that?

I would think we'd say:

```
I can read, change or delete OAuth applications when:
...
I am the admin of the organization where the application was created
...
```

Also, `I am the user of the application.` is listed under read/change/delete... is that strictly true? a normal user can't delete any application which they are using right? That only implies that they can use it to get a token?

Reply:
Verify that in the code below this also... it almost looks like anyone who can see the application could also modify/remove it.

Reply:
Think more strict rules can be applied after the introduction of more granular permissions.

Reply:
At the moment, the application doesn't appear tied to a specific organization, so how would you determine which organization admins had access to it?

Reply:
As long as applications still belong to users, the RBAC of an application can be determined by the underlying user.